src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,298 @@
+import { PdfInfoIF, PresignedUrlIF } from '@/interfaces'
+
+// set web worker
+GlobalWorkerOptions.workerSrc = require('pdfjs-dist/build/pdf.worker.entry')
+
+@Component({})
+export default class FileUploadPdf extends Vue {
+  @Prop({ default: null }) readonly file!: File
+  @Prop({ default: null }) readonly fileKey!: string
+  @Prop({ default: true }) readonly isRequired!: boolean
+  @Prop({ default: 0 }) readonly maxSize!: number // in MB
+  @Prop({ default: null }) readonly pageSize!: PageSizes
+  @Prop({ required: true }) readonly userId!: string
+
+  // Network service functions
+  @Prop({ required: true })
+  readonly getPresignedUrl!: (fileName: string) => Promise<PresignedUrlIF>
+  @Prop({ required: true })
+  readonly uploadToUrl!: (url: string, file: File, key: string, userId: string) => Promise<AxiosResponse>
+
+  /** Component key, used to force it to re-render. */
+  protected count = 0
+
+  /** Custom errors messages, use to put component into manual error mode. */
+  protected errorMessages = [] as Array<string>
+
+  /** Clears data and local state. */
+  public reset (): void {
+    // update parent
+    this.updateFile(null)
+    this.updateFileKey(null)
+
+    // clear local state
+    this.errorMessages = []
+
+    // force re-render to clear file input
+    this.count++
+  }
+
+  /** Resets input validation only. */
+  public resetValidation (): void {
+    this.errorMessages = []
+  }
+
+  /**
+   * "Shallow validates" this component.
+   * "Deep validation" is done when a file is selected.
+   */
+  public validate (): boolean {
+    // force re-render to clear file input
+    this.count++
+
+    // Logic is as follows:
+    // if (no file) and (required) => error
+    // if (no file) and (not required) => valid
+    // if (have file) and (have key) => valid
+    // if (have file) and (no key) => error
+
+    if (!this.file) {
+      if (this.isRequired) {
+        this.errorMessages = ['File is required']
+        return false
+      } else {
+        this.errorMessages = []
+        return true
+      }
+    }
+
+    // if we have a file key then file is valid and was uploaded
+    if (this.fileKey) return true
+
+    return false
+  }
+
+  /** When file is selected or cleared, validates the file and uploads it. */
+  protected async onChange (file: File): Promise<void> {
+    // update parent for later reactivity
+    this.updateFile(file)
+    this.updateFileKey(null)
+
+    // Logic is as follows:
+    // if (no file) and (required) => error
+    // if (no file) and (not required) => valid
+    // if (have file) and (required) => depends on file validation
+    // if (have file) and (not required) => depends on file validation
+
+    if (!file) {
+      if (this.isRequired) {
+        this.errorMessages = ['File is required']
+        return
+      } else {
+        this.errorMessages = []
+        return
+      }
+    }
+
+    // validate the file
+    this.errorMessages = ['Processing...']
```

Comment:
New: show this (and line 126) while validating and uploading the PDF, which can take a few seconds.

This was approved by Ethan.